src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/HtmlFormattingTest.cs

```diff
@@ -749,6 +749,91 @@ public async Task FormatNestedComponents2()
             tagHelpers: GetComponents());
     }
 
+    [Fact]
+    public async Task FormatNestedComponents3()
+    {
+        await RunFormattingTestAsync(
+            input: """
+                    @if (true)
+                    {
+                        <Component1 Id="comp1"
+                                Caption="Title" />
+                    <Component1 Id="comp2"
+                                Caption="Title">
+                                <Frag>
+                    <Component1 Id="comp3"
+                                Caption="Title" />
+                                </Frag>
+                                </Component1>
+                    }
+                    """,
+            expected: """
+                    @if (true)
+                    {
+                        <Component1 Id="comp1"
+                                    Caption="Title" />
+                        <Component1 Id="comp2"
+                                    Caption="Title">
+                            <Frag>
+                                <Component1 Id="comp3"
+                                            Caption="Title" />
+                            </Frag>
+                        </Component1>
+                    }
+                    """,
+            tagHelpers: GetComponents());
+    }
+
+    [Fact]
+    public async Task FormatNestedComponents4()
+    {
+        await RunFormattingTestAsync(
+            input: """
+                    @{
+                        RenderFragment fragment =
+                          @<Component1 Id="Comp1"
+                                     Caption="Title">
+                        </Component1>;
+                    }
+                    """,
+            expected: """
+                    @{
+                        RenderFragment fragment =
+                        @<Component1 Id="Comp1"
+                                     Caption="Title">
+                        </Component1>;
+                    }
```

Comment:
Would this be the expected formatting?

Reply:
I think this is fine, but render fragments are weird. See https://github.com/dotnet/razor/issues/6632
I suspect this will end up with the `Caption` attribute starting one character to the left of where `Id` is, because the `@` only pushes out the first line, rather than setting an indentation baseline for the whole fragment. Changing that would be outside the scope of this PR I think 🙂